src/api/app/models/obs_factory/staging_project.rb

```diff
@@ -245,17 +245,23 @@ def fetch_status
       @checks = []
       @missing_checks = []
       repositories.each do |repo|
-        build_id = repo.build_id
-        status = repo.status_reports.find_by(uuid: build_id)
-        if status
-          @missing_checks += status.missing_checks
-          @checks += status.checks
-        else
-          @missing_checks += repo.required_checks
+        add_status(repo)
+        repo.repository_architectures.each do |repo_arch|
+          add_status(repo_arch)
         end
       end
     end
 
+    def add_status(checkable)
+      status = checkable.status_reports.latest
+      if status
+        @missing_checks += status.missing_checks
+        @checks += status.checks
+      else
+        @missing_checks += checkable.required_checks
+      end
```

Comment:
Hm, I don't like too much that we need to distinguish here if it exists or not ... Maybe we should always have a status report or move it to the checkables? But just food for thoughts ...

Reply:
I'd actually prefer the same. But it's a little tricky

Reply:
Yeah I can imagine ... won't block the PR though, maybe we need to see where this is going first and then do a little refactoring.